src/modules/setup.ts

```diff
@@ -0,0 +1,277 @@
+              await outputJsonSync(packageJsonPath, packageJson, { spaces: '\t' })  // Revert package.json to original state.
+            }
+
+          }
+        }
+      }
+    ),
+    values,
+    Promise.all
+  )(buildersWithAppDeps as Record<string, any>)
+
+}
+
+const tsconfig = async (account: string, workspace: string) => {
+  const http = builderHttp(account, workspace)
+  const retryOpts = {
+    retries: 2,
+    minTimeout: 1000,
+    factor: 2,
+  }
+  const downloadTSConfig = async (_: any, tryCount: number) => {
+    if (tryCount > 1) {
+      log.info(`Retrying...${tryCount-1}`)
+    }
+    try {
+      const res = await http.get(`/_v/builder/0/tsconfig`)
+      return res.data
+    } catch (err) {
+      const statusMessage = err.response.status ?
+        `: Status ${err.response.status}` : ''
+      log.error(`Error fetching tsconfig from builder-hub ${statusMessage} (try: ${tryCount})`)
+      throw err
+    }
+  }
+  try {
+    return retry(downloadTSConfig, retryOpts)
+  } catch (e) {
+    log.error('Unable to get tsconfig.json info from vtex.builder-hub.')
+    return {}
+  }
+}
+
+export const getTSConfig = async (manifest: Manifest, account: string, workspace: string) => {
+  const tsconfigsFromBuilder = await tsconfig(account, workspace)
+  const buildersWithBaseTSConfig =
+    compose(
+      ramdaReject(isNil),
+      mapObjIndexed(
+        (version: string, builder: string) => { const builderTSConfig = prop(builder, tsconfigsFromBuilder)
+          if (builderTSConfig && has(version, builderTSConfig)) {
+            return prop(version, builderTSConfig)
+          }
+          return null
+        }
+      ),
+      prop('builders')
+    )(manifest)
+
+
+  return mapObjIndexed(
+    (baseTSConfig: any, builder: any) => {
+      try {
+      const tsconfigPath = resolveTSConfigPath(builder)
+      const currentTSConfig = readJsonSync(tsconfigPath)
+      const newTSConfig = mergeDeepRight(currentTSConfig, baseTSConfig)
+      outputJsonSync(tsconfigPath, newTSConfig, { spaces: 2 })  // Revert package.json to original state.
+      } catch(e) {
+        log.error(e)
+      }
+    }
+  )(buildersWithBaseTSConfig as Record<string, any>)
+}
+
+export const setupTSLint = async (manifest: Manifest) => {
+  const builders = keys(prop('builders', manifest) || {})
+  const filteredBuilders = intersection(builders, buildersToAddAdditionalPackages)
+  const lintDeps = keys(addToPackageJson)
+  compose<any, any, any>(
+    Promise.all,
+    map(
+      async (builder: string) => {
+        try {
+          const packageJsonPath = resolvePackageJsonPath(builder)
+          const devDependencies = (prop('devDependencies', await readJson(packageJsonPath))) || {}
+          if (difference(lintDeps, intersection(lintDeps, keys(devDependencies))).length !== 0) {
+            yarnAddESLint(builder)
+            await outputJson(resolveTSLintPath(builder), addToEslintrc[builder], { spaces: 2 })
```

Comment:
this line shouldn't be outside the if statement?

Reply:
as it is, it only changes the user's package.json if it is missing one or more of the dependencies (`lintDeps`) we want to add.

Reply:
1. but this is related to `.eslintrc`. If my `devDependencies` doesn't includes `lintDeps` my `.eslintrc` will be overriden. If my `devDependencies` includes `lintDeps` my `.eslintrc` will not be overriden. Why not to always override `.eslintrc`?
2. `resolveTSLintPath` could be renamed to `resolveESLintPath` also

Reply:
1. Ok! As we agreed, changed the behavior to always overwrite the user's `.eslintrc`.
2. Thx!